explainaboard/analysis/analyses.py

```diff
@@ -307,6 +317,37 @@ def perform(
         )
 
 
+@dataclass(frozen=True)
+class ComboOccurence:
+    """A struct representing occurences of the string tuples.
+
+    Args:
+        features: The feature values that the occurence is counted.
+        sample_ids: List of sample IDs that contain the feature.
+    """
+
+    features: tuple[str, ...]
+    sample_count: int
+    sample_ids: list[int]
+
+    @staticmethod
+    def from_dict(dikt: dict) -> ComboOccurence:
+        """Deserialization method."""
+        return ComboOccurence(
+            features=dikt['features'],
+            sample_count=dikt['sample_count'],
+            sample_ids=dikt['sample_ids'],
+        )
+
+    def __lt__(self, other: ComboOccurence):
+        """Implement__lt__ to allow natural sorting."""
+        return (self.features, self.sample_count, self.sample_ids) < (
+            other.features,
+            self.sample_count,
+            other.sample_ids,
+        )
```

Comment:
Randomized variables is not suitable for comparison, and it may cause unintended behavior.

```suggestion
        return (self.features, self.sample_count) < (
            other.features,
            other.sample_count
        )
```

Reply:
Resolved. Thanks a lot! 

Just out of curiosity, why would that be a bug?

Reply:
> why would that be a bug?

Because the RHS contained `self.sample_count`

Reply:
Fixed. Thanks!